### Problem 1
$f=2x_{1}^{2} - 4x_1 x_2+ 1.5x^{2}_{2}+ x_2$

Find the stationary point (Gradient = 0):

$
G=\begin{bmatrix}
\frac{\partial f}{\partial x_1}  \\
\frac{\partial f}{\partial x_2}
\end{bmatrix}
=\begin{bmatrix}
4x_1-4x_2  \\
-4x_1+3x_2+1
\end{bmatrix}=0 \\\\
$

$
\therefore
\begin{align}
4x_1-4x_2=0 \\
-4x_1+3x_2+1=0 \\
x_1=x_2 \\
-x_1 = -1 \\
\end{align}
$

$\therefore $ Stationary point:

$\\ x=\begin{bmatrix} 1 \\ 1 \end{bmatrix}$

Find the Hessian matrix at the stationary point:

$
H=\begin{bmatrix}
\frac{\partial^2 f}{\partial x_1^2} & \frac{\partial^2 f}{\partial x_1 \partial x_2} \\
\frac{\partial^2 f}{\partial x_2 \partial x_1} & \frac{\partial^2 f}{\partial x_2^2}
\end{bmatrix}
=\begin{bmatrix}
4 & -4 \\
-4 & 3
\end{bmatrix} \\\\
$


$|I\lambda-H|=0 \\\\
\begin{vmatrix}
\lambda-4 & 4 \\
4 & \lambda-3
\end{vmatrix} = (\lambda-4)(\lambda-3)-16=0 \\\\
\lambda^2-7\lambda-4=0
$

In [4]:
import numpy as np
print(np.roots([1,-7,-4]))

[ 7.53112887 -0.53112887]


$\lambda_1 = 7.53112887 \\\\
\lambda_2 = -0.53112887$
$\therefore$ H matrix has both positive and negative eigenvalues, the stationary point is a saddle point
